### ISSUE TITLE
ServiceStack ASP.NET with Razor

### ISSUE THREAD

**Issue by jhallam3** [ServiceStack/ServiceStackVS#31]
Hi, 

Im trying to use the 'ServiceStack ASP.NET with Razor' option however in VS2015 SP2 it doesn't work.
I get this screenshot attached. 

![screen shot 2016-07-11 at 08 47 00](https://cloud.githubusercontent.com/assets/5174328/16725001/400c7af8-474d-11e6-8dcf-994c37beb202.png)

**Comment by Layoric** [comment]
@jhallam3 that's strange, I've created new projects on two separate machines running VS 2015 Update 2 and the project builds and runs successfully.

Could you provide VS info by going to `Help`->`About` and click `Copy Info` and paste into this issue? This might help narrowing down what might be causing the issue.

Cheers.

**Comment by jhallam3** [comment]
HI,
Listed below.


Microsoft Visual Studio Community 2015
Version 14.0.25123.00 Update 2
Microsoft .NET Framework
Version 4.6.01055

Installed Version: Community

Microsoft Visual Studio Tools for Applications 2015   00322-20000-00000-AA014
Microsoft Visual Studio Tools for Applications 2015

Visual Basic 2015   00322-20000-00000-AA014
Microsoft Visual Basic 2015

Visual C# 2015   00322-20000-00000-AA014
Microsoft Visual C# 2015

Visual C++ 2015   00322-20000-00000-AA014
Microsoft Visual C++ 2015

Windows Phone SDK 8.0 - ENU   00322-20000-00000-AA014
Windows Phone SDK 8.0 - ENU

Application Insights Tools for Visual Studio Package   5.209.60613.1
Application Insights Tools for Visual Studio

ASP.NET and Web Tools 2015.1 (Beta8)   14.1.11106.0
ASP.NET and Web Tools 2015.1 (Beta8)

ASP.NET Web Frameworks and Tools 2012.2   4.1.41102.0
For additional information, visit http://go.microsoft.com/fwlink/?LinkID=309563

ASP.NET Web Frameworks and Tools 2013   5.2.40314.0
For additional information, visit http://www.asp.net/

Common Azure Tools   1.7
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

GorillaPlayerPackage Extension   1.0
GorillaPlayerPackage Visual Studio Extension Detailed Info

JavaScript Language Service   2.0
JavaScript Language Service

JavaScript Project System   2.0
JavaScript Project System

Microsoft Azure Mobile Services Tools   1.4
Microsoft Azure Mobile Services Tools

MySQL for Visual Studio   1.2.6
Data design and management tools for MySQL.  Copyright © 2007-2015 Oracle, Inc.

Node.js Tools   1.1.40329.04
Adds support for developing and debugging Node.js apps in Visual Studio

Node.js Tools - Profiling   1.1.40329.04
Profiling support for Node.js projects.

NuGet Package Manager   3.4.4
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

PreEmptive Analytics Visualizer   1.2
Microsoft Visual Studio extension to visualize aggregated summaries from the PreEmptive Analytics product.

ServiceStackVS   1.0
Templates and tools for use with the ServiceStack framework.

SQL Server Data Tools   14.0.60311.1
Microsoft SQL Server Data Tools

TypeScript   1.8.31.0
TypeScript tools for Visual Studio

Visual Studio Tools for Universal Windows Apps   14.0.25219.00
The Visual Studio Tools for Universal Windows apps allow you to build a single universal app experience that can reach every device running Windows 10: phone, tablet, PC, and more. It includes the Microsoft Windows 10 Software Development Kit.

Xamarin   4.1.0.530 (2e39740)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin.Android   6.1.0.71 (4e27558)
Visual Studio extension to enable development for Xamarin.Android.

Xamarin.iOS   9.8.0.323 (39ebb77)
Visual Studio extension to enable development for Xamarin.iOS.

**Comment by Layoric** [comment]
Nothing is standing out, are there any errors listed in the `ServiceStackVS` output window regarding NuGet? Were there any errors during the creation of the template?

**Comment by jhallam3** [comment]
nothing jumped out. I can make a video to show u what Im doing. will that help?

**Comment by Layoric** [comment]
That might help, but before you go to the trouble, could you please paste the `ServiceStackVS` output window contents after you try to create a project from the `ServiceStack ASP.NET with Razor` template? Should just be logs regarding installing NuGet dependencies if everything was successful.

**Comment by jhallam3** [comment]
please see the video 

http://screencast.com/t/QbZpHx3vtowP

below is the output for servicestackVS 

--- Installing latest ServiceStack NuGet dependencies for 'RazorWebApp1' ---
--- Finished installing latest ServiceStack NuGet dependencies for 'RazorWebApp1' ---
--- Installing latest ServiceStack NuGet dependencies for 'RazorWebApp1.ServiceModel' ---
--- Finished installing latest ServiceStack NuGet dependencies for 'RazorWebApp1.ServiceModel' ---
--- Installing latest ServiceStack NuGet dependencies for 'RazorWebApp1.ServiceInterface' ---
--- Finished installing latest ServiceStack NuGet dependencies for 'RazorWebApp1.ServiceInterface' ---
--- Installing latest ServiceStack NuGet dependencies for 'RazorWebApp1.Tests' ---
--- Finished installing latest ServiceStack NuGet dependencies for 'RazorWebApp1.Tests' ---


Jon

**Comment by Layoric** [comment]
@jhallam3 Ahh I see, I misunderstood the problem. I am also seeing the same issue with Razor intellisense with VS 2015 Update 2. 

I've tried some of the following things listed in [this StackOverflow answer](http://stackoverflow.com/questions/31581666/visual-studio-2015-broken-razor-intellisense) but still having the same issue myself, however does seem to work for some people so might be worth a try.

I'll update here if I find a fix. Let me know also if you manage to get the Intellisense working for VS 2015.

**Comment by kdhannan** [comment]
I am also having the same problems and I have been able to resolve all but 1 error.

I added the version and public key information in the web.config to clear most of the errors:: 
```
<configuration>
  <configSections>
    <sectionGroup name="system.web.webPages.razor" type="System.Web.WebPages.Razor.Configuration.RazorWebSectionGroup, System.Web.WebPages.Razor, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35">
      <section name="host" type="System.Web.WebPages.Razor.Configuration.HostSection, System.Web.WebPages.Razor, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35" requirePermission="false" />
      <section name="pages" type="System.Web.WebPages.Razor.Configuration.RazorPagesSection, System.Web.WebPages.Razor, Version=3.0.0.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35" requirePermission="false" />
    </sectionGroup>
  </configSections>
</configuration>

<system.web.webPages.razor>
    <host factoryType="System.Web.Mvc.MvcWebRazorHostFactory, System.Web.Mvc, Version=5.2.3.0, Culture=neutral, PublicKeyToken=31BF3856AD364E35" />
</system.web.webPages.razor>
```

This left me with the System.Web.Helpers reference error so I added this reference from Assemblies > Extensions.

The last error is `The name 'Context' does not exist in the current context` and I have absolutely no idea how to get rid of this one.

Hope that helps!

**Comment by mythz** [comment]
If this is a Razor intelli-sense error it doesn't have any impact on the behavior of the page, it's a result of VS.NET intelli-sense being coupled to MVC's implementation of Razor.

But you should be able to clear this design error by downloading the latest [packages on MyGet](https://github.com/ServiceStack/ServiceStack/wiki/MyGet) and your pages should start with an `@inherits` declaration, e.g:

```csharp
@inherits ViewPage<ResponseDto>
```

**Comment by Layoric** [comment]
@kdhannan This has been resolved in the latest SSVS + ServiceStack 4.0.62 release, thanks for reporting! :+1:
